nemo/collections/tts/models/base.py

```diff
@@ -28,9 +28,38 @@
 from nemo.core.neural_types.neural_type import NeuralType
 from nemo.utils import logging, model_utils
 
+PYNINI_AVAILABLE = True
+try:
+    import nemo_text_processing
+except (ImportError, ModuleNotFoundError):
+    PYNINI_AVAILABLE = False
 
-class SpectrogramGenerator(ModelPT, ABC):
-    """ Base class for all TTS models that turn text into a spectrogram """
+
+class NeedsNormalizer(ModelPT, ABC):
+    """Base class for all TTS models that needs text normalization(TN)"""
+
+    def _setup_normalizer(self, cfg):
+        if "text_normalizer" in cfg:
```

Comment:
Nitpick: It might be cleaner to have `_setup_normalizer()` directly in the `SpectrogramGenerator`, with a constructor argument that lets the model class decide whether it is required or not. But not required, since most of this is legacy code anyways.